### PR TITLE
Use wire-type compatible source in TinyGo

### LINKED ISSUE #13: Generate go source from the same protos used by the scheduler framework
**What would you like to be added**:

Makefile machinery to generate kubernetes/proto/api source from k8s.io/api@v0.26.2/core/v1/generated.proto instead of our kubernetes/proto/kubernetes.proto.

**Why is this needed**:

kubernetes/proto/kubernetes.proto  isn't compatible with the proto in the scheduler framework. It both has different field order and in some cases different fields. This means manual data translation if we are to use it.

**Completion requirements**:

This enhancement requires the following artifacts:

- [ ] Design doc
- [x] API change
- [ ] Docs update

The artifacts should be linked in subsequent comments.

### PR DESCRIPTION
**Pull request by codefromthecrypt**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Before, we had incompatible types, which among other things, made the POC difficult to reason with from a performance point of view. We were manually converting fields, which would never work in practice as the model is quite large.

This uses the real protos k8s does, just generating TinyGo compatible guest code from them. This now serializes the `v1.Pod` directly from the input type in a way TinyGo can get it. To eliminate a performance hit from filters that don't access the current pod, this extracts a `FilterArgs` type. Doing so gives us flexibility in implementation, especially as we add more parameters later.

#### Which issue(s) this PR fixes:

Fixes #13

#### Special notes for your reviewer:

This change slows down execution, but that's good because it is more realistic. Before we manually copied a single field, so the *after* is a better baseline. Note: this still doesn't use realistic data, but at least we can generate it now!
```bash
$ cd plugin
$ go test -run='^$' -bench '^BenchmarkFilter$' . -count=6 > old.txt
$ go test -run='^$' -bench '^BenchmarkFilter$' . -count=6 > new.txt
$ benchstat old.txt new.txt
goos: darwin
goarch: arm64
pkg: sigs.k8s.io/kube-scheduler-wasm-extension/plugin
                                                    │   old.txt   │              new.txt               │
                                                    │   sec/op    │   sec/op     vs base               │
Filter/success:_node_is_match_with_spec.NodeName-12   12.62µ ± 0%   19.14µ ± 1%  +51.73% (p=0.002 n=6)
```

### CODE REVIEW

**Review by codefromthecrypt — COMMENTED** [review]
note: using realistic protos increases the base wasm binary size in tinygo to 756KB. Future work may be able to optimize this back without losing features, but it is worth saying this part now.

**Comment by codefromthecrypt** [comment]
ps here is the output from the makefile. Note there are warnings about schema not being in use, but as they are _ imported you would have to change the source to exclude it.

```bash
$ make update-kubernetes-proto
cd ./kubernetes/proto/tools; \
	cat tools.go | grep "_" | awk -F'"' '{print $2}' | xargs -tI % go install %
go install github.com/knqyf263/go-plugin/cmd/protoc-gen-go-plugin
go install google.golang.org/protobuf/cmd/protoc-gen-go
echo "Regenerate the Go protobuf code."
Regenerate the Go protobuf code.
cd kubernetes/kubernetes/staging/src/; \
	protoc ./k8s.io/apimachinery/pkg/api/resource/generated.proto --go-plugin_out=../../../proto \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/api/resource/generated.proto=./resource; \
	protoc ./k8s.io/apimachinery/pkg/runtime/generated.proto --go-plugin_out=../../../proto \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/generated.proto=./runtime; \
	protoc ./k8s.io/apimachinery/pkg/runtime/schema/generated.proto --go-plugin_out=../../../proto \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/schema/generated.proto=./schema; \
	protoc ./k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto --go-plugin_out=../../../proto \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/apis/meta/v1/generated.proto=./meta \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/runtime \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/schema/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/schema; \
	protoc ./k8s.io/apimachinery/pkg/util/intstr/generated.proto --go-plugin_out=../../../proto \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/util/intstr/generated.proto=./instr; \
	protoc ./k8s.io/api/core/v1/generated.proto --go-plugin_out=../../../proto \
		--go-plugin_opt=Mk8s.io/api/core/v1/generated.proto=./api \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/api/resource/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/resource \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/apis/meta/v1/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/meta \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/runtime \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/runtime/schema/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/schema \
		--go-plugin_opt=Mk8s.io/apimachinery/pkg/util/intstr/generated.proto=sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/instr;
k8s.io/apimachinery/pkg/apis/meta/v1/generated.proto:25:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
k8s.io/api/core/v1/generated.proto:27:1: warning: Import k8s.io/apimachinery/pkg/runtime/schema/generated.proto is unused.
```

**Review by sanposhiho — APPROVED** [review]
One nit, leave it to @kerthcet.

/approve

File: Makefile [hunk 1]
```diff
@@ -11,14 +11,31 @@ submodule-update:
...
 .PHONY: update-kubernetes-proto
 update-kubernetes-proto: proto-tools
-	echo "You need to install protoc before running this."
-	echo "Regenerate the protobuf definition from the submodule ./kubernetes/kubernetes."
-	openapi2proto -spec ./kubernetes/kubernetes/api/openapi-spec/swagger.json -out ./kubernetes/proto/kubernetes.proto
 	echo "Regenerate the Go protobuf code."
-	protoc ./kubernetes/proto/kubernetes.proto \
-		--go-plugin_out=./kubernetes/proto --go-plugin_opt=Mkubernetes/proto/kubernetes.proto=./api
+	cd kubernetes/kubernetes/staging/src/; \
+	protoc ./k8s.io/apimachinery/pkg/api/resource/generated.proto --go-plugin_out=../../../proto \
```
Comment (kerthcet) [inline Makefile:23]: I did this same before, it's annoying at this moment, but I prefer to use the native ones.

**Review by kerthcet — COMMENTED** [review]
/lgtm
Thanks @codefromthecrypt Have a great flight!

**Comment by kerthcet** [comment]
Please squash you commit it said you have invalid commit message.

**Comment by codefromthecrypt** [comment]
ok I tried a different commit message 🤞

**Comment by codefromthecrypt** [comment]
Thanks for the fast feedback. now we are unlocked to do more modeling and performance work!

**Pull request merged**